cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/externalprocess/ExternalProcessHandler.java

```diff
@@ -525,7 +491,22 @@ public boolean isNotConnected() {
         try {
             getConnection("GET", "Content-Type", "text/plain", ExternalProcessConstants.IS_CONNECTION_READY);
             if (conn.getResponseCode() < 300) {
-                return false;
+
+                // Check if it is the correct server version
+                BufferedReader br = new BufferedReader(new InputStreamReader(conn.getInputStream()));
+                String response = br.readLine();
+
+                if (response.equals(processProperties.getServerVersion())) {
+                    return false;
+                }
+
+                if (response.equals("true")) {
+                    LOG.warn(
+                        "An old version is currently deployed. Please consider deploying the newest version to get the current bug fixes/features");
```

Comment:
Exactly this should be done automatically.
There is a download function at the moment, which automatically download the nestserver, but does not store it as far as I know including the version number.
This should be kept. So in case a previous version of the tsplugin for example downloads nestserver-1.0.5.exe and then you run a new instance of the tsplugin, which depends on nestserver-1.0.7.exe. In this case automatically the new version should be downloaded as well and should automatically be run.

Reply:
Let me explain the logic here briefly. E.g the tsplugin wants to use nestserver. The first thing it does is check if it can connect to a particular port where the server should be running (in this case 5000). If there is an old version of nestserver running on that port, it will use that version and does not trigger any download. As that the new version will not get downloaded at all. Also even if it is somehow got downloaded to the local folder, it will never be started.

Reply:
ok, smart. And in case it's not the correct version, would cobigen try to download automatically the new version?
This would be necessary for the fix for me.

Reply:
Exactly. In case cobigen finds out that it is connecting to the wrong server version, it will download the version defined in pom and start it

Reply:
great. One last question: there was a time, when the nestserver exe was not downloaded and stored including a version suffix. Meaning it was downloading for example nestserver-1.0.0.exe from the web, but simply stored nestserver.exe on the filesystem, which would break the ability to have multiple versions of the nestserver next to each other.
Could you once please check that this is not the case?

Reply:
The nestserver exe should always contains the version. If it is not, it's properly a broken release of nestserver itself. Cobigen does not rename the downloaded file. The reason why there are no multiple versions of nestserver in the past is because in the old logic, cobigen cleans up all the old versions of nestserver if it downloads a new one.